src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue25124.cs

```diff
@@ -0,0 +1,22 @@
+using NUnit.Framework;
+using UITest.Appium;
+using UITest.Core;
+
+namespace Microsoft.Maui.TestCases.Tests.Issues;
+
+public class Issue25124 : _IssuesUITest
+{
+	public Issue25124(TestDevice testDevice) : base(testDevice)
+	{
+	}
+
+	public override string Issue => "OnPlatform does not work in the Header and Footer of CollectionView";
+
+	[Test]
+	[Category(UITestCategories.CollectionView)]
+	public void ShouldDisplayHeaderBasedOnOnPlatform()
+	{
+		App.WaitForElement("CollectionView");
+		VerifyScreenshot();
```

Comment:
Pending snapshots. Already available in the latest build.

Example, Mac:
![image](https://github.com/user-attachments/assets/74cd90c7-bb06-440e-a410-6a32a3b477d6)

Could you commit the images?